core/bin/zksync_tee_prover/src/tee_prover.rs

```diff
@@ -188,15 +188,15 @@ impl Task for TeeProver {
                     }
                     retries += 1;
                     tracing::warn!(%err, "Failed TEE prover step function {retries}/{}, retrying in {} milliseconds.", self.config.max_retries, backoff.as_millis());
-                    tokio::time::timeout(backoff, stop_receiver.0.changed())
-                        .await
-                        .ok();
                     backoff = std::cmp::min(
                         backoff.mul_f32(self.config.retry_backoff_multiplier),
                         self.config.max_backoff,
                     );
                 }
             }
+            tokio::time::timeout(backoff, stop_receiver.0.changed())
+                .await
+                .ok();
```

Comment:
This will sleep for 1s even if there are consecutive batches (e.g `Some` after `Some`), and I don't think we want that.
Probably you want to handle `Ok(Some(_))`, `Ok(None)` and `Err(_)` differently (e.g. don't wait if we actually processed a batch). Simplest option would be to return `needs_to_sleep` from `match result`, but you will get bonus points if you'll implement a more elegant solution :)

Reply:
Fixed. ✅ Not sure how to implement this better. If we want to parallelize it later, we'll probably have to redesign it anyway.